lib/spack/spack/cmd/install.py

```diff
@@ -177,7 +177,10 @@ def default_log_file(spec):
     """
     fmt = 'test-{x.name}-{x.version}-{hash}.xml'
     basename = fmt.format(x=spec, hash=spec.dag_hash())
-    dirname = fs.os.path.join(spack.paths.var_path, 'junit-report')
+    dirname = spack.config.get(
+        'config:misc_cache', spack.paths.var_path)
```

Comment:
I'm not sure we want to clear this every time we clear the misc cache.  Reports are really less transient than what we usually put in there.

Can you make it so that these have their own directory within `~/.spack`?  I'd recommend putting them in `~/.spack/reports/junit`, and adding `~/.spack/reports` to `spack.paths`.

Reply:
@tgamblin makes sense,  done! The only downside is that this directory location (`~/.spack/reports/junit`) is now no longer configurable, right? But perhaps that is not required anyway.